packages/patrol_cli/lib/src/commands/build_ios.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:path/path.dart' show basename;
+import 'package:patrol_cli/src/base/exceptions.dart';
+import 'package:patrol_cli/src/base/extensions/core.dart';
+import 'package:patrol_cli/src/base/logger.dart';
+import 'package:patrol_cli/src/crossplatform/app_options.dart';
+import 'package:patrol_cli/src/dart_defines_reader.dart';
+import 'package:patrol_cli/src/ios/ios_test_backend.dart';
+import 'package:patrol_cli/src/pubspec_reader.dart';
+import 'package:patrol_cli/src/runner/patrol_command.dart';
+import 'package:patrol_cli/src/test_finder.dart';
+
+class BuildIOSCommand extends PatrolCommand {
+  BuildIOSCommand({
+    required TestFinder testFinder,
+    required DartDefinesReader dartDefinesReader,
+    required PubspecReader pubspecReader,
+    required IOSTestBackend iosTestBackend,
+    required Logger logger,
+  })  : _testFinder = testFinder,
+        _dartDefinesReader = dartDefinesReader,
+        _pubspecReader = pubspecReader,
+        _iosTestBackend = iosTestBackend,
+        _logger = logger {
+    usesTargetOption();
```

Comment:
I like this pattern. But I see it comes from `PatrolCommand`. IMO this should be separate from the parent-class of all commands, in something like `FlutterBuildingPatrolCommandMixin`...? This name may suck, but you get the idea. `PatrolCommand` could of course preserve the arg parsing capabilities, but leave the `usesXOption` to more usecase group-oriented mixins

Reply:
TBH, I took inspiration from `flutter_tools`. They also have `FlutterCommand` and `usesXOption` methods are defined in there as well.

I agree with you, but I think it's... kind of an overkill?